src/services/dbtTestsService.ts

```diff
@@ -0,0 +1,28 @@
+import { provideSingleton } from "../utils";
+import { MacroMetaMap } from "../domain";
+
+@provideSingleton(DbtTestService)
+export class DbtTestService {
+  // Find the file path of test macro
+  public getMacroFilePath = (
+    macros: [string],
+    projectName: string,
+    macroMetaMap: MacroMetaMap,
+    testName: string | undefined,
+  ) => {
+    if (!testName) {
+      return;
+    }
+
+    // Find if current test depends on test macro in current project
+    const macro = macros.find(
+      (m) => m === `macro.${projectName}.test_${testName}`,
+    );
+
+    if (macro) {
+      // return the file path if it ends with sql
+      const macroData = macroMetaMap.get(`test_${testName}`);
+      return macroData?.path.endsWith(".sql") ? macroData?.path : undefined;
+    }
+  };
+}
```

Comment:
Can we not add this code in the macroParser? This way people will still need to remember to inject this class every time we us `macro.path`

Reply:
Actually, macros in macroMetamap has the right path values, but not the tests in testsMetamap. We could move this to testParser, but we dont have macroMetamap available there. Is there a way to get macro data in testParser? @mdesmet